owltorrent-network/src/main/java/edu/rice/owltorrent/network/messages/BitfieldMessage.java

```diff
@@ -0,0 +1,31 @@
+package edu.rice.owltorrent.network.messages;
+
+import edu.rice.owltorrent.network.PeerMessage;
+import java.nio.ByteBuffer;
+import java.util.BitSet;
+
+public class BitfieldMessage extends PeerMessage {
+
+  public static final int BASE_SIZE = 5;
+  public BitSet bitfield;
+
+  public BitfieldMessage(MessageType type, BitSet bitfield) {
+    super(type);
+    this.bitfield = bitfield;
+  }
+
+  @Override
+  public byte[] toBytes() {
+    ByteBuffer buffer = ByteBuffer.allocate(BASE_SIZE + this.bitfield.size() / 8);
```

Comment:
I'm not sure how bitset size works, but /8 is floored. Is this correct when the size isn't a multiple of 8? For example 9 bit/8 = 1 byte, whereas you want 2 bytes instead.

Reply:
It will always be a multiple of 8 as the payload of Bitfield is always padded to complete bytes.